CONTRIBUTING.md

```diff
@@ -1,106 +1,186 @@
+
+- `dotnet msbuild eng\mgmt.proj /t:Build /p:scope=Compute`
+- `dotnet msbuild eng\mgmt.proj /t:RunTests /p:scope=Compute`
+- `dotnet msbuild eng\mgmt.proj /t:CreateNugetPackage /p:scope=Compute`
 
 ### Update build tools
-Build tools are now downloaded as part of a nuget package under root\restoredPackages\microsoft.internal.netsdkbuild.mgmt.tools
-If for any reason there is an update to the build tools, you will then need to first delete directory root\restoredPackages\microsoft.internal.netsdkbuild.mgmt.tools and reexecute your build command. This will simply get the latest version of build tools.
 
+Build tools are now downloaded as part of a nuget package under `root\restoredPackages\microsoft.internal.netsdkbuild.mgmt.tools`
+If for any reason there is an update to the build tools, you will then need to first delete directory `root\restoredPackages\microsoft.internal.netsdkbuild.mgmt.tools` and re-execute your build command. This will simply get the latest version of build tools.
 
-## To on-board new libraries
+# On-boarding New Libraries
 
 ### Project Structure
 
-In "sdk\< Service Name >", you will find projects for services that have already been implemented
+In `sdk\< Service Name >`, you will find projects for services that have already been implemented
 
-  - The easiest way is to model your test/sdk project using existing test/sdk project
-    - Make a copy of existing csproj and change the meta data relevant to your SDK
-    - This ensures the SDK that will be generated will be consistent (including consistent target frameworks)
-  - Each service contains a project for their generated/customized code
-    - The folder 'Generated' contains the generated code
-    - The folder 'Customizations' contains additions to the generated code - this can include additions to the generated partial classes, or additional classes that augment the SDK or 
+1. Client library projects needs to use the $(RequiredTargetFrameworks) *(defined in eng/Directory.Build.Data.props)* property in its TargetFramework while management library projects should use $(SdkTargetFx) _(defined in AzSdk.reference.props)_
+2. Projects of related packages are grouped together in a folder following the structure specified in [Repo Structure](https://github.com/Azure/azure-sdk/blob/master/docs/engineering-system/repo-structure.md)
+   - Client library packages are in a folder name like **_Microsoft.Azure.< ServiceName >_**
+   - Management library packages are in a folder named like **_Microsoft.Azure.Management.< Resource Provider Name >_**
+3. Each shipping package contains a project for their **generated** and /or **customized** code
+   - The folder **'Generated'** contains the generated code
+   - The folder **'Customizations'** or **'Customized'** contains additions to the generated code - this can include additions to the generated partial classes, or additional classes that augment the SDK or call the generated code
```

Comment:
Any reason customized is being mentioned, we should push for one name "customization", it helps in making everything consistent.